tools/io-trace-explorer/io_trace.h

```diff
@@ -0,0 +1,151 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+/*
+ * io_trace_kind --
+ *     The type (kind) of the trace.
+ */
+enum class io_trace_kind {
+    DEVICE,
+    FILE,
+    WIRED_TIGER,
+};
+
+/*
+ * io_trace_operation --
+ *     A single data point within a trace.
+ */
+struct io_trace_operation {
+
+    double timestamp; /* The timestamp in seconds, relative to the start of the trace. */
+    char action;      /* The action, as defined by blktrace, plus a few custom actions. */
+
+    bool read;        /* Was this a read? */
+    bool write;       /* Was this a write? */
+    bool synchronous; /* Whether this was a synchronous operation. */
+    bool barrier;     /* Whether this included a barrier operation. */
+    bool discard;     /* Whether this was a discard operation. */
+
+    unsigned long offset; /* Offset in bytes from the beginning of the file or the device. */
+    unsigned length;      /* Length in bytes. */
+    double duration;      /* Duration in seconds, if available (or 0 if not). */
+    char process[32];     /* The issuing process name, if available. */
+
+    /*
+     * wrap_timestamp --
+     *     Wrap a timestamp into an instance of io_trace_operation, which is useful for filtering a
+     * collection of traces.
+     */
+    static io_trace_operation
+    wrap_timestamp(double t)
+    {
+        io_trace_operation r;
+        bzero(&r, sizeof(r));
```

Comment:
I was unfamiliar with `bzero`, upon googling it appears to set the memory of the given struct to all zeros. I believe it is a deprecated and POSIX only. In C++ the standard way to zero a struct is: `r = {}`.